misc/formatter/modules/formatter-cli/src/main/java/org/ballerinalang/formatter/cli/FormatUtil.java

```diff
@@ -0,0 +1,358 @@
+            if (!dryRun) {
+                // Write formatted content to the file.
+                FormatUtil.writeFile(fileName, formattedSource);
+            }
+            formattedFiles.add(fileName);
+        }
+    }
+
+    private static List<String> iterateAndFormat(BLangPackage bLangPackage, Path sourceRootPath, boolean dryRun)
+            throws IOException {
+        List<String> formattedFiles = new ArrayList<>();
+
+        // Iterate compilation units and format.
+        for (BLangCompilationUnit compilationUnit : bLangPackage.getCompilationUnits()) {
+            formatAndWrite(compilationUnit, sourceRootPath, formattedFiles, dryRun);
+        }
+
+        // Iterate testable packages and format.
+        for (BLangTestablePackage testablePackage : bLangPackage.getTestablePkgs()) {
+            for (BLangCompilationUnit compilationUnit : testablePackage.getCompilationUnits()) {
+                formatAndWrite(compilationUnit, sourceRootPath, formattedFiles, dryRun);
+            }
+        }
+
+        return formattedFiles;
+    }
+
+    /**
+     * Get prepared compiler context.
+     *
+     * @param sourceRootPath ballerina compilable source root path
+     * @return {@link CompilerContext} compiler context
+     */
+    private static CompilerContext getCompilerContext(Path sourceRootPath) {
+        CompilerPhase compilerPhase = CompilerPhase.DEFINE;
+        CompilerContext context = new CompilerContext();
+        CompilerOptions options = CompilerOptions.getInstance(context);
+        options.put(PROJECT_DIR, sourceRootPath.toString());
+        options.put(OFFLINE, Boolean.toString(false));
+        options.put(COMPILER_PHASE, compilerPhase.toString());
+        options.put(SKIP_TESTS, Boolean.toString(false));
+        options.put(TEST_ENABLED, "true");
+        options.put(LOCK_ENABLED, Boolean.toString(false));
+        options.put(EXPERIMENTAL_FEATURES_ENABLED, Boolean.toString(true));
+        options.put(PRESERVE_WHITESPACE, Boolean.toString(true));
+        options.put(NEW_PARSER_ENABLED, Boolean.toString(false));
+
+        return context;
+    }
+
+    /**
+     * Check whether the given module name exists.
+     *
+     * @param module      module name
+     * @param projectRoot path of the ballerina project root
+     * @return {@link Boolean} true or false
+     */
+    private static boolean isModuleExist(String module, Path projectRoot) {
+        Path modulePath;
+        if (module.startsWith("src/")) {
+            modulePath = projectRoot.resolve(module);
+        } else {
+            modulePath = projectRoot.resolve("src").resolve(module);
+        }
+
+        return modulePath.toFile().isDirectory();
+    }
+
+    /**
+     * Check whether the given path isn't a source root of a ballerina project.
+     *
+     * @param path - path where the command is executed from
+     * @return {@link boolean} true or false
+     */
+    private static boolean notABallerinaProject(Path path) {
```

Comment:
Better function name would be `isBallerinaProject` instead of negative reply.

Reply:
Done